src/gsy_e/models/strategy/energy_parameters/heatpump/heat_pump.py

```diff
@@ -236,25 +234,29 @@ def update_tanks_temperature(
 
         self._hp_state.set_net_heat_consumed_kJ(last_time_slot, net_energy_kJ)
 
-    def update_cop(
+    def update_cop_after_dis_charging(
         self,
         source_temp_C: float,
         time_slot: DateTime,
         last_time_slot: DateTime,
+        bought_energy_kWh: float,
     ):
         """Update the COP of the heat pump in its state class."""
-        heat_demand_kJ = self._hp_state.get_heat_demand_kJ(last_time_slot)
-        if heat_demand_kJ < FLOATING_POINT_TOLERANCE:
+        if bought_energy_kWh < FLOATING_POINT_TOLERANCE:
             cop = self._hp_state.get_cop(last_time_slot)
         else:
-            cop = self._calc_cop(heat_demand_kJ, source_temp_C, last_time_slot)
+            cop = self._calc_cop(
+                produced_heat_kJ=convert_kWh_to_kJ(bought_energy_kWh),
```

Comment:
Thank you! This indeed was a bug in the reporting of the cop which was important to solve.